src/main/java/picard/annotation/SortGff.java

```diff
@@ -0,0 +1,269 @@
+public class SortGff extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Sorts a gff3 file, and adds flush directives";
+    static final String USAGE_DETAILS = "<h3> Summary </h3>\n" +
+            "  <p> This tool sorts a gff3 file by coordinates, so that it can be indexed.\n" +
+            " It additionally adds flush directives where possible, which can significantly reduce the memory footprint of downstream tools.\n" +
+            " Sorting of multiple contigs can be specified by a sequence dictionary; if no sequence dictionary is specified, contigs are sorted lexicographically. </p>\n" +
+            "\n" +
+            " <h3> Usage Examples </h3>\n" +
+            " <h4> 1. Sort gff3 file, add flush directives.  Contigs will be sorted lexicographically.</h4>\n" +
+            " <pre>\n" +
+            " java -jar picard.jar SortGff\n" +
+            "      I=input.gff3\n" +
+            "      O=output.gff3\n" +
+            " </pre>\n" +
+            "\n" +
+            " <h4> 2. Sort gff3 file, add flush directives.  Contigs will be sorted according to order of sequence dictionary</h4>\n" +
+            " <pre>\n" +
+            " java -jar picard.jar SortGff\n" +
+            "      I=input.gff3\n" +
+            "      O=output.gff3\n" +
+            "      SD=dictionary.dict\n" +
+            " </pre>";
+
+    @Argument(doc = "Input Gff3 file to sort.", shortName = StandardOptionDefinitions.INPUT_SHORT_NAME)
+    public File INPUT;
+
+    @Argument(doc = "Sorted Gff3 output file.", shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME)
+    public File OUTPUT;
+
+    @Argument(doc = "Dictionary to sort contigs by.  If dictionary is not provided, contigs are sorted lexicographically.", shortName = StandardOptionDefinitions.SEQUENCE_DICTIONARY_SHORT_NAME, optional = true)
+    public File SEQUENCE_DICTIONARY;
+
+    @Argument(doc = "Number of records to hold in memory before spilling to disk", optional = true)
+    public int nRecordsInMemory = 50000;
+
+    private final Log log = Log.getInstance(SortGff.class);
+
+    private final Map<String, Integer> latestStartMap = new HashMap<>();
+    private int latestStart = 0;
+    String latestChrom;
+
+    private static class FeatureComparator implements Comparator<Gff3Feature> {
+        final SAMSequenceDictionary dict;
+
+        FeatureComparator(final SAMSequenceDictionary dict) {
+            this.dict = dict;
+        }
+
+        public int compare(final Gff3Feature f1, final Gff3Feature f2) {
+            int comp = dict == null ? f1.getContig().compareTo(f2.getContig()) : dict.getSequenceIndex(f1.getContig()) - dict.getSequenceIndex(f2.getContig());
+            if (comp == 0) {
+                comp = f1.getStart() - f2.getStart();
+            }
+
+            return comp;
+        }
+    }
+
+    @Override
+    protected int doWork() {
+        final Gff3Codec inputCodec = new Gff3Codec(Gff3Codec.DecodeDepth.SHALLOW);
```

Comment:
consider adding input and output checking before computing
`IOUtil.assertFileIsReadable(INPUT);
  IOUtil.assertFileIsWritable(OUTPUT);`